clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -553,6 +610,13 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
     }
 
 #ifdef UNIX
+    if (saw_rseq_abort) {
+        shard->saw_rseq_abort_ = true;
+    } else if (!(memref.marker.type == TRACE_TYPE_MARKER &&
+                 (memref.marker.marker_type == TRACE_MARKER_TYPE_TIMESTAMP ||
+                  memref.marker.marker_type == TRACE_MARKER_TYPE_CPU_ID))) {
```

Comment:
What is the logic for clearing on anything except a timestamp/cpuid?

Reply:
To detect whether the current signal caused an rseq abort we need to find the following pattern:
```
rseq_abort
kernel_event
kernel_event
```

I found that sometimes the two kernel events are separated by timestamp+cpu.

Reply:
Added a comment to clarify.